lib/changelog/pipedream.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Changelog.Pipedream do
+  @moduledoc """
+  Pass in a URL or schema struct to `purge` it from Pipedream's cache
+  """
+  alias Changelog.HTTP
+
+  def host, do: Application.get_env(:changelog, :pipedream_host)
+  def port, do: Application.get_env(:changelog, :pipedream_port) |> String.to_integer()
+  def scheme, do: Application.get_env(:changelog, :pipedream_scheme)
+  def purge_token, do: Application.get_env(:changelog, :pipedream_purge_token)
+
+  def purge(url) do
+    case Changelog.Dns.aaaa(host()) do
+      {:ok, addresses} ->
+        addresses
+        |> Task.async_stream(
+          fn address ->
+            case HTTP.request(:purge, purge_url(url, address), "", purge_headers(url)) do
+              {:ok, _response} ->
+                :ok
```

Comment:
I think that we need to fail if the response is `401 Unauthorized`.

I just hit a misconfiguration where the purge token was not being set correctly. The function returned `:ok` but the purge didn't actually work.

WDYT @jerodsanto ?

Reply:
Agreed. I addressed this in 4ae8a701